File: abc_dz_2/abc_dz2_source/classes/car.cpp
//
// Created by Дмитрий Татьянченко on 18.10.2021.
//

#include "car.h"

Car::Car(int fuel_tank_capacity, double fuel_consumption, short speed_limit) :
        Vehicle(fuel_tank_capacity, fuel_consumption) {

    this->type_ = (Vehicle::CAR);

    if (speed_limit < 0) {
        throw std::invalid_argument("speed limit < 0");
    }

    this->speed_limit_ = speed_limit;
}

std::string Car::toString() {
    return "Car: " + this->Vehicle::toString() +
           " speed limit: " + std::to_string(this->speed_limit_);
}

File: abc_dz_3/abc_dz3_source/vehicle/car.py
from vehicle.vehicle import *


class Car(Vehicle):
    def __init__(self, fuel_tank_capacity, fuel_consumption, speed_limit):
        super().__init__(fuel_tank_capacity, fuel_consumption)
        self.__speed_limit = speed_limit

    def to_string(self):
        return "CAR " + super().to_string() + " speed limit (mph): " + str(self.__speed_limit)
